src/FontManager.cpp

```diff
@@ -61,12 +61,38 @@ void FontManager::loadFonts(const QString& folder)
     QDir dir = folder;
     dir.setNameFilters(filters);
 
-    foreach (QString fontfile, dir.entryList(QDir::Files | QDir::Readable | QDir::NoDotAndDotDot)) {
-        QString fontFilePathName = QStringLiteral("%1/%2").arg(dir.absolutePath(), fontfile);
-        if (QFontDatabase::addApplicationFont(fontFilePathName) == -1) {
-            // At the point that addFonts() is called we have a GUI application
-            // in the making and can use qDebug() and not rely on iostream class
-            qWarning() << "FontManager::loadFonts() ERROR - Could not load the font in the file: " << fontFilePathName;
-        }
+    foreach (QString fontFile, dir.entryList(QDir::Files | QDir::Readable | QDir::NoDotAndDotDot)) {
+        QString fontFilePathName = QStringLiteral("%1/%2").arg(dir.absolutePath(), fontFile);
+        loadFont(fontFilePathName);
+    }
+}
+
+void FontManager::loadFont(const QString& filePath) {
+    if (fontAlreadyLoaded(filePath)) {
+        return;
     }
+
+    auto fontID = QFontDatabase::addApplicationFont(filePath);
+
+    // remember even if the font failed to load so we don't spam messages on fonts that repeat
+    rememberFont(filePath, fontID);
+
+    if (fontID == -1) {
+        qWarning() << "FontManager::loadFonts() warning - Could not load the font in the file: " << filePath;
+    }
+}
+
+bool FontManager::fontAlreadyLoaded(const QString& filePath) {
+    QFileInfo fontFile(filePath);
+    auto fileName = fontFile.fileName();
+
+    return loadedFonts.contains(fileName);
+}
+
+void FontManager::rememberFont(const QString& filePath, int fontID) {
+    QFileInfo fontFile(filePath);
+    auto fileName = fontFile.fileName();
+
+    if (loadedFonts.contains(fileName)) { return; }
```

Comment:
:-1: Please no! Conditional code block on the same line as the condition - that's like trying to [parse XHTML with a Regexp](https://stackoverflow.com/a/1732454/4805858)... 😉